hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -639,8 +639,8 @@ private static HoodieMetadataColumnStats mergeColumnStatsRecords(HoodieMetadataC
 
     Comparable maxValue =
         (Comparable) Stream.of(
-            (Comparable) unwrapStatisticValueWrapper(prevColumnStats.getMinValue()),
-            (Comparable) unwrapStatisticValueWrapper(newColumnStats.getMinValue()))
+            (Comparable) unwrapStatisticValueWrapper(prevColumnStats.getMaxValue()),
+            (Comparable) unwrapStatisticValueWrapper(newColumnStats.getMaxValue()))
```

Comment:
Nice catch, is there any possibility we write a unit test for this clazz ?

Reply:
+1

Reply:
Added the tests